charts/aad-pod-identity/templates/identities.yaml

```diff
@@ -1,24 +1,26 @@
-{{- if .Values.azureIdentity.enabled }}
-apiVersion: "aadpodidentity.k8s.io/v1"
-kind: AzureIdentity
-metadata:
-  name: {{ .Values.azureIdentity.name }}
-  namespace: {{ template "aad-pod-identity.azureidentity.namespace" . }}
-  labels:
-    {{- include "aad-pod-identity.labels" . | nindent 4 }}
-spec:
-  type: {{ .Values.azureIdentity.type }}
-  resourceID: {{ required ".Values.azureIdentity.resourceID is required!" .Values.azureIdentity.resourceID }}
-  clientID: {{ required ".Values.azureIdentity.clientID is required!" .Values.azureIdentity.clientID }}
----
-apiVersion: "aadpodidentity.k8s.io/v1"
-kind: AzureIdentityBinding
-metadata:
-  name: {{ .Values.azureIdentityBinding.name }}
-  namespace: {{ template "aad-pod-identity.azureidentitybinding.namespace" . }}
-  labels:
-    {{- include "aad-pod-identity.labels" . | nindent 4 }}
-spec:
-  azureIdentity: {{ .Values.azureIdentity.name }}
-  selector: {{ required ".Values.azureIdentityBinding.selector is required!" .Values.azureIdentityBinding.selector }}
+apiVersion: v1
```

Comment:
We should add a length check for `.Values.azureIdentities` in case the user didn't supply any identities.

Reply:
Good catch.

Reply:
Is the relation between binding and identity always 1:1? Shouldn’t we separate it in two lists in the templates?

Reply:
I think 1:1 makes sense. Each `AzureIdentity` should only have one `AzureIdentityBinding`, which can be consumed by multiple pods.

Reply:
1:1 makes sense in _most_ scenarios, but there must be a reason to have two separate CRDs in the first place. Which legitimates the question whether we should define them as 1 in the configuration and templates. I don’t mind going forward with the current implementation, but I think it’s a valid question in itself.

What are your thoughts, @aramase ?

Reply:
If I remember correctly from testing it, we'll just have an empty list of identities if the user didn't supply any. Nonetheless, let me add this validation.

Reply:
@aristosvo Keeping it 1:1 makes sense. It makes it easier for users to configure multiple ids and bindings all at once.